pkg/apis/extensions/v1beta1/types.go

```diff
@@ -1036,3 +1037,58 @@ type PodSecurityPolicyList struct {
 	// Items is a list of schema objects.
 	Items []PodSecurityPolicy `json:"items"`
 }
+
+// Template contains the inputs needed to produce a Config.
+type Template struct {
```

Comment:
That is how this concept is referred to in the [proposal](https://github.com/kubernetes/kubernetes/blob/release-1.2/docs/proposals/templates.md).  Feedback on naming would be helpful as always.  Perhaps ApplicationTemplate?